lib/common/patchset.c

```diff
@@ -1081,11 +904,26 @@ pcmk__xml_apply_patchset(xmlNode *xml, const xmlNode *patchset,
     }
 
     if (rc == pcmk_ok) {
+        /* @COMPAT We might receive a v1 patchset if Pacemaker was built with
+         * deprecated symbols, a user creates a v1 patchset with
+         * xml_create_patchset(1, ...), and the user applies the patchset.
+         *
+         * We could also receive a v1 patchset if a *different* node's Pacemaker
+         * was built with deprecated symbols and a user sends a
+         * PCMK__CIB_REQUEST_APPLY_PATCH. However, that's enough of an edge case
+         * that it seems okay to throw an "Unknown patch format" error unless
+         * this node's Pacemaker allows deprecated symbols.
+         *
+         * Otherwise, there's no reason we should ever encounter a v1 patchset.
+         * Pacemaker has generated v2 patchsets internally since 1.1.4.
+         */
         crm_element_value_int(patchset, PCMK_XA_FORMAT, &format);
         switch (format) {
+#if !defined(PCMK_ALLOW_DEPRECATED) || (PCMK_ALLOW_DEPRECATED == 1)
             case 1:
                 rc = pcmk_rc2legacy(apply_v1_patchset(xml, patchset));
                 break;
+#endif  // !defined(PCMK_ALLOW_DEPRECATED) || (PCMK_ALLOW_DEPRECATED == 1)
```

Comment:
PCMK_ALLOW_DEPRECATED is only for headers, not source files. Pacemaker is always *built* with deprecated symbols, the constant just controls whether they are exposed to source that includes the headers. Internal code can't see the deprecated symbols, external code can -- but it's only the visibility that's controlled. If we didn't build the deprecated symbols, they wouldn't be deprecated but removed, which would break API/ABI compatibility.

Reply:
Oh, duh. Yeah that makes total sense. I guess I was thinking about people building Pacemaker with deprecated symbols allowed/disallowed, but that's not where they'd control it -- they'd control it in their own external programs.